### PR TITLE
make internal solution for usaco snakes gold

### PR DESCRIPTION
**Pull request by TruaShamu**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as
done._

- [x] I have tested my code
- [x] I have followed the code style guidelines mentioned
      [here](https://usaco.guide/general/adding-solution/#code-conventions)
- [x] I have properly formatted the files according to the steps mentioned
      [here](https://usaco.guide/general/adding-solution#steps)

### CODE REVIEW

**Comment by TruaShamu** [comment]
Proof that it works:
blehhh

File: solutions/usaco-945.mdx [hunk 1]
```diff
@@ -0,0 +1,73 @@
+---
+id: usaco-945
+source: USACO Gold 2019 US Open
+title: Snakes
+author: Sofia Yang
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_snakes_gold_open19.html)
```
Comment (dutinmeow) [inline solutions/usaco-945.mdx:8]:
```suggestion
[Official Analysis and Code(C++)](http://www.usaco.org/current/data/sol_snakes_gold_open19.html)
```

**Review by dutinmeow — APPROVED** [review]

File: solutions/usaco-945.mdx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+**Time Complexity:** $\mathcal{O}(N^3)$
+
+<LanguageSection>
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class snakes {
```
Comment (SansPapyrus683) [inline solutions/usaco-945.mdx:21]:
```suggestion
public class Snakes
```

File: solutions/usaco-945.mdx [hunk 3]
```diff
@@ -0,0 +1,73 @@
+---
+id: usaco-945
+source: USACO Gold 2019 US Open
+title: Snakes
+author: Sofia Yang
+---
+
+[Official Analysis and Code(C++)](http://www.usaco.org/current/data/sol_snakes_gold_open19.html)
```
Comment (SansPapyrus683) [inline solutions/usaco-945.mdx:8]:
```suggestion
[Official Analysis and Code (C++)](http://www.usaco.org/current/data/sol_snakes_gold_open19.html)
```

File: solutions/usaco-945.mdx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+		int groups = io.nextInt();
+		int changes = io.nextInt();
+
+		/*
+		 * dp[i][j] is the min sum of net sizes used
+		 * to pick up the first i groups with j net size changes.
+		 */
+		int[][] dp = new int[groups + 1][changes + 1];
+		int[] inp = new int[groups + 1];
+		int noWaste = 0; // Just the sum of snakes in each group.
```
Comment (SansPapyrus683) [inline solutions/usaco-945.mdx:33]: two spaces after the code before starting the `//`

File: solutions/usaco-945.mdx [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+		/*
+		 * The space wasted is just the current space
+		 * used minus the space used if there is unlimited size changes.
+		 */
+		int answer = dp[groups][changes] - noWaste;
+		io.println(answer);
+		io.close();
+
+	}
+	//CodeSnip{Kattio}
```
Comment (SansPapyrus683) [inline solutions/usaco-945.mdx:68]:
```suggestion
	
	//CodeSnip{Kattio}
```

File: solutions/usaco-945.mdx [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+		Kattio io = new Kattio("snakes");
+		int groups = io.nextInt();
+		int changes = io.nextInt();
+
+		/*
+		 * dp[i][j] is the min sum of net sizes used
+		 * to pick up the first i groups with j net size changes.
+		 */
+		int[][] dp = new int[groups + 1][changes + 1];
+		int[] inp = new int[groups + 1];
```
Comment (SansPapyrus683) [inline solutions/usaco-945.mdx:32]: inp? maybe name it something like `snakes` instead?

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-945.mdx (hunk 3 above)
Comment (ryanchou-dev) [reply solutions/usaco-945.mdx:8]: waait isn't is supposed to just be `[Official Analysis (C++)]`

File: solutions/usaco-945.mdx (hunk 2 above)
Comment (ryanchou-dev) [reply solutions/usaco-945.mdx:21]: did you mean to delete the opening brace?

File: solutions/usaco-945.mdx [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+		 * used minus the space used if there is unlimited size changes.
+		 */
+		int answer = dp[groups][changes] - noWaste;
+		io.println(answer);
+		io.close();
+
+	}
+	//CodeSnip{Kattio}
+}
+
```
Comment (ryanchou-dev) [inline solutions/usaco-945.mdx:70]:
```suggestion
```

**Review by ryanchou-dev — APPROVED** [review]
looks good!

File: solutions/usaco-945.mdx [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+			for (int j = 1; j <= changes; j++) {
+				dp[i][j] = Integer.MAX_VALUE;
+				int size = snakes[i];
+				// Changing the net size.
+				for (int prev = i - 1; prev >= 0; prev--) {
+					/* The cost is the previous cost,
+					 * plus the cost of picking up the groups from prev to i.
+					 */
+					dp[i][j] = Integer.min(dp[i][j],
+										dp[prev][j - 1] + size * (i - prev));
```
Comment (SansPapyrus683) [inline solutions/usaco-945.mdx:52]: indentation sus
also why `Integer.min`?

File: solutions/usaco-945.mdx [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+			dp[i][0] = max * i;
+
+			for (int j = 1; j <= changes; j++) {
+				dp[i][j] = Integer.MAX_VALUE;
+				int size = snakes[i];
+				// Changing the net size.
+				for (int prev = i - 1; prev >= 0; prev--) {
+					/* The cost is the previous cost,
+					 * plus the cost of picking up the groups from prev to i.
+					 */
```
Comment (SansPapyrus683) [inline solutions/usaco-945.mdx:50]: this comment is formatted kinda weird

**Review by jessechoe10 — APPROVED** [review]

**Review by SansPapyrus683 — APPROVED** [review]

**Review by nathangong — APPROVED** [review]
looks good

**Pull request merged**